modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/ScriptManagerBridge.cs

```diff
@@ -615,7 +615,7 @@ internal static godot_bool TryReloadRegisteredScriptWithClass(IntPtr scriptPtr)
                 Type? top = scriptType;
                 Type native = GodotObject.InternalGetClassNativeBase(top);
 
-                while (top != null && top != native)
+                if (top != null && top != native)
                 {
                     var methodList = GetMethodListForType(top);
```

Comment:
If we are removing the loop, we can replace `top` here with `scriptType` which should never be null so no need to check.

```suggestion
                if (scriptType != native)
                {
                    var methodList = GetMethodListForType(scriptType);
```